pkg/cmd/ssh/options.go

```diff
@@ -757,17 +757,18 @@ func waitForBastion(ctx context.Context, o *SSHOptions, gardenClient client.Clie
 			return false, err
 		}
 
-		cond := corev1alpha1helper.GetCondition(bastion.Status.Conditions, operationsv1alpha1.BastionReady)
-
-		if cond == nil || cond.Status != gardencorev1alpha1.ConditionTrue {
-			lastCheckErr = errors.New("bastion does not have BastionReady=true condition")
+		switch cond := corev1alpha1helper.GetCondition(bastion.Status.Conditions, operationsv1alpha1.BastionReady); {
+		case cond == nil:
+			return false, nil
+		case cond.Status != gardencorev1alpha1.ConditionTrue:
+			lastCheckErr = errors.New(cond.Message)
 			fmt.Fprintf(o.IOStreams.ErrOut, "Still waiting: %v\n", lastCheckErr)
 			return false, nil
 		}
 
 		lastCheckErr = bastionAvailabilityChecker(preferredBastionAddress(bastion), privateKeyBytes)
 		if lastCheckErr != nil {
-			fmt.Fprintf(o.IOStreams.ErrOut, "Still waiting: cannot connect to bastion yet: %v\n", lastCheckErr)
+			fmt.Fprintf(o.IOStreams.ErrOut, "Still waiting: Bastion is ready, waiting to establish an ssh connection with bastion: %v\n", lastCheckErr)
```

Comment:
```suggestion
			fmt.Fprintf(o.IOStreams.ErrOut, "Still waiting: Bastion is ready, however cannot connect to bastion yet: %v\n", lastCheckErr)
```
sorry for the late response. I tested it again and I feel there are too many "waiting"s in the log line. I prefer the old error message together with the "Bastion is ready" information that you have also added

Reply:
changed 575dbf071b14a7a39b7676608feae411a67b1eba,  
one thing I let to share, in CCEE internal environment. the error message is repeated for a long time due to the CCEE network, security rules taking time to activate, or other reasons, hope people have patination to wait. 
I haven't figured out why the ssh slowness, SRE internal ticket was created want to speed the ssh process up.

```
Still waiting: bastion does not have BastionReady=true condition
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Still waiting: cannot connect to bastion yet: dial tcp xx.xxx.xxx.xx:22: connect: connection refused
Bastion host became available at xx.xxx.xxx.xx.
You can open additional SSH sessions using the command below:
```